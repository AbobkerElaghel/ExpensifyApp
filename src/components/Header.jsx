import React from "react";
import {NavLink} from "react-router-dom";

const Header = () => (
    <header>
        <h1>ExpensifyApp</h1>
        <p> <NavLink activeClassName="is-active-link" exact to="/">Home Page </NavLink></p>
        <p> <NavLink activeClassName="is-active-link" to="/create">Add Page </NavLink></p>
        <p> <NavLink activeClassName="is-active-link" to="/edit">Edit Page </NavLink></p>
        <p> <NavLink activeClassName="is-active-link" to="/help">Help Page </NavLink></p>
    </header>
);

export default Header;