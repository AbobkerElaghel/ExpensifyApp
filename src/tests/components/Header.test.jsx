import React from "react";
import { shallow } from "enzyme";
import Header from "../../components/Header.jsx";

test("should render Header component correctly", () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot()
    // const renderer = new ReactShallowRenderer();
    // renderer.render(<Header />);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
});