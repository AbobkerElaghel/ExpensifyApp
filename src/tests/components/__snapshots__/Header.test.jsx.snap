// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component correctly 1`] = `
<header>
  <h1>
    ExpensifyApp
  </h1>
  <p>
     
    <NavLink
      activeClassName="is-active-link"
      exact={true}
      to="/"
    >
      Home Page 
    </NavLink>
  </p>
  <p>
     
    <NavLink
      activeClassName="is-active-link"
      to="/create"
    >
      Add Page 
    </NavLink>
  </p>
  <Connect(Logout) />
</header>
`;
